src/common/containers/with-repository.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+
+import { setGitHubRepository } from '../actions/repository-input';
+
+function withRepository(WrappedComponent) {
```

Comment:
New HOC component that contains bit of functionality to propagate `repository` reducer with data (needed for wrapped component).

It requires `fullName` prop (in most cases it's filled from URL params) and `repository` that is taken from redux store.

I'm not sure if it's the best way to implement this (it still requires `connect` layer on top to provide `fullName` and `repository`), but it seems to do the trick (removed duplicated code from repository setup, builds list and build details containers).